ci/scripts/python_build.sh

```diff
@@ -75,5 +75,6 @@ popd
 
 if [ "${BUILD_DOCS_PYTHON}" == "ON" ]; then
   ncpus=$(python -c "import os; print(os.cpu_count())")
+  pip install -r ${arrow_dir}/docs/requirements.txt
```

Comment:
Should be already installed in the docker images. Otherwise we should rather fix the images.

Reply:
There were build errors when this line wasn't there. So I guess the build image didn't contain this dependencies. 
See https://github.com/apache/arrow/runs/4469906684?check_suite_focus=true#step:8:3987

Reply:
> @jorisvandenbossche do we want to include this in 7.0? Personally I'd like to :)

It would be great if we can include this in version 7.0
I see it as a good improvement to docs and I feel it would sad to not shipping it due to some areas for improvement here and there compared to the current docs that has types that don't work at all.

Reply:
Fixed it.